src/frontend/efiling-frontend/src/components/page/rush/Rush.js

```diff
@@ -1,42 +1,357 @@
 import { getSidecardData } from "../../../modules/helpers/sidecardData";
 import Payment from "../../../domain/payment/Payment";
 
 import "./Rush.scss";
+import { getCountries } from "./RushService";
+import RushDocumentList from "./rush-document-list/RushDocumentList";
+import { Toast } from "../../toast/Toast";
+import { Input } from "../../input/Input";
 
-const calloutText = `Please provide the date of when the direction was made,
-the name of the Judge who made the direction along with any additional details
-you feel are necessary.`;
+const calloutText = `Please provide the date of when the direction was made, the name of the Judge who made the direction along with any additional details you feel are necessary.  `;
 
 const generateInputField = (input, onChange) => (
   <div className="form-child">
     <Input input={input} onChange={onChange} />
   </div>
 );
 
+const formatPhoneNumber = (phoneNumber) => {
+  if (!phoneNumber) {
+    return phoneNumber;
+  }
+
+  const phoneNumberDigits = phoneNumber.replace(/[^\d]/g, "");
+
+  if (phoneNumberDigits.length < 4) {
+    return phoneNumberDigits;
+  }
+
+  if (phoneNumberDigits.length < 7) {
+    return `${phoneNumberDigits.slice(0, 3)}-${phoneNumberDigits.slice(3)}`;
+  }
+
+  return `${phoneNumberDigits.slice(0, 3)}-${phoneNumberDigits.slice(
+    3,
+    6
+  )}-${phoneNumberDigits.slice(6, 10)}`;
+};
+
 export default function Rush({ payment }) {
+  // eslint-disable-next-line no-unused-vars
+  const [continueBtnEnabled, setContinueBtnEnabled] = useState(false);
+
   const input = {
     isReadOnly: false,
     styling: "bcgov-editable-white",
     isRequired: true,
   };
   const aboutCsoSidecard = getSidecardData().aboutCso;
   const csoAccountDetailsSidecard = getSidecardData().csoAccountDetails;
+  const rushSubmissionSidecard = getSidecardData().rushSubmission;
+  const contactMethods = [
+    ["Email", "email"],
+    ["Phone Number", "phoneNumber"],
+  ];
+  const clearFields = {
+    surname: null,
+    firstName: null,
+    contactMethod: contactMethods[0],
+    phoneNumber: null,
+    email: null,
+    org: null,
+    country: null,
+    details: null,
+  };
 
+  const [files, setFiles] = useState([]);
   const [selectedDate, setSelectedDate] = useState(new Date());
   const [showPayment, setShowPayment] = useState(false);
+  const [radio1, setRadio1] = useState(false);
+  const [radio2, setRadio2] = useState(false);
+  const [radio3, setRadio3] = useState(false);
+  const [fields, setFields] = useState(clearFields);
+  const [numDocumentsError, setNumDocumentsError] = useState(false);
+  const [duplicateFilenamesError, setDuplicateFilenamesError] = useState(false);
+  const [emailError, setEmailError] = useState(null);
+  const [toastMessage, setToastMessage] = useState("");
+  const [showToast, setShowToast] = useState(false);
+  const [countries, setCountries] = useState([]);
+
+  const checkForDuplicateFilenames = (droppedFiles, previousFiles) => {
```

Comment:
Another one that might make more sense as a helper module since it can be reused for other screens?

Reply:
Since this is already used on multiple screens that's a good call. We should refactor this into a helper module.